kv.go

```diff
@@ -124,6 +124,17 @@ type Iterator interface {
 	Close() error
 }
 
+// Dropable is an optionall interface implemented by some Stores.
+//
+// It provides a convienient and cheap way of deleting all the data in an existing store.
+type Dropable interface {
+	// DropAll deletes all the data stored in the store.
+	//
+	// Concurrent writes will be blocked, however depending on the underlying store implementation,
+	// concurrent reads may not be.  Badger does not block concurrent reads, but Memory store does.
+	DropAll() error
```

Comment:
Suggestion: Slight preference for `Reset` over `DropAll` terminology.

Reply:
I dislike `Reset` as I do not feel that is strongly enough conveys that calling it will delete all of the data in the store (highly destructive).

Reply:
How about `DeleteAll`?

Reply:
What do you not like about `DropAll`?  It is what is used by badger, and it's name does not limit itself to key-values only in the same way that `DeleteAll` seems to.

Reply:
I just don't feel `Drop` indicates irreversible `Deletion`. Always makes me wonder, can I `pick` back the dropped stuff?

Like with git for example, you can always cherry pick back dropped commits, etc.

I understand, though, that this is a badger terminology.

Reply:
> Like with git for example, you can always cherry pick back dropped commits, etc.

Is a nice example, thanks :)

Reply:
It's not just a badger terminology though. It's a database terminology. SQL uses it to delete rows and tables. It's not a reversible action. I don't think DB admins will be confused by the terminology.

Reply:
I'm going to leave the name as is, it feels *very* unimportant to me atm.  In the off chance that the name causes confusion for the protocol guys or whoever in the future it should be very easy to change - is not like this function is going to ever be referenced in a million different places.

Reply:
Yes, it's common now among DB terminologies. Nonetheless, I still think there is better terminology that can be used.

Is a non-blocking suggestion, regardless. Haha